com.unity.multiplayer.mlapi/Tests/Runtime/RpcQueueTests.cs

```diff
@@ -78,33 +47,160 @@ public IEnumerator RpcQueueUnitTest()
+
+            Assert.IsTrue(testsAreComplete && testsAreValidated);
+
+            //Disable this so it isn't running any longer.
+            rpcPipelineTestComponent.gameObject.SetActive(false);
+
+            yield return null;
+#else
+            yield return null;
+#endif
+        }
+
+        /// <summary>
+        /// This tests the RPC Queue outbound and inbound buffer capabilities.
+        /// It will send
+        /// </summary>
+        /// <returns>IEnumerator</returns>
+        [UnityTest]
+        public IEnumerator BufferDataValidation()
+        {
+ // Disabling this test on 2019.4 due to ILPP issues on Yamato CI/CD runs (UNITY_2020_2_OR_NEWER)
+ // Adding the ability to test if we are running in editor local (UNIT_MANUAL_TESTING)
+#if UNITY_2020_2_OR_NEWER || UNITY_MANUAL_TESTING
+            Assert.IsTrue(NetworkManagerHelper.StartNetworkManager());
 
-            //Stop pinging
-            rpcPipelineTestComponent.PingSelfEnabled = false;
-            Debug.Log("RPC Queue Testing completed.");
+            Guid gameObjectId = NetworkManagerHelper.AddGameNetworkObject("GrowingBufferObject");
 
-            //Stop the host
-            NetworkManager.Singleton.StopHost();
+            var growingRpcBufferSizeComponent = NetworkManagerHelper.AddComponentToObject<BufferDataValidationComponent>(gameObjectId);
+            NetworkManagerHelper.SpawnNetworkObject(gameObjectId);
 
-            //Shutdown the NetworkManager
-            NetworkManager.Singleton.Shutdown();
+            //Start Testing
+            growingRpcBufferSizeComponent.EnableTesting = true;
 
-            Debug.Log($"Exiting status => {nameof(testsAreComplete)}: {testsAreComplete} - {nameof(testsAreValidated)}: {testsAreValidated} - {nameof(instantiatedNetworkManager)}: {instantiatedNetworkManager} - {nameof(exceededMaximumStageIterations)}: {exceededMaximumStageIterations}");
+            var testsAreComplete = growingRpcBufferSizeComponent.IsTestComplete();
 
-            Assert.IsTrue(testsAreComplete && testsAreValidated && instantiatedNetworkManager);
+            //Wait for the rpc pipeline test to complete or if we exceeded the maximum iterations bail
+            while (!testsAreComplete)
+            {
+                //Wait for 20ms
+                yield return new WaitForSeconds(0.02f);
 
-            GameObject.DestroyImmediate(playerObject);
-            GameObject.DestroyImmediate(networkManagerObject);
+                testsAreComplete = growingRpcBufferSizeComponent.IsTestComplete();
+            }
+
+            //Stop Testing
+            growingRpcBufferSizeComponent.EnableTesting = false;
+
+            //Just disable this once we are done.
+            growingRpcBufferSizeComponent.gameObject.SetActive(false);
+
+            Assert.IsTrue(testsAreComplete);
+
+            yield return null;
```

Comment:
no need for this in this block, we're already `yield`ing in the while-loop above.
```suggestion
```